gamla/graph.py

```diff
@@ -23,15 +23,22 @@ def graph_traverse_many(
 
     Note: `get_neighbors` must return elements without duplicates."""
     queue = [*sources]
-    seen = set(map(key, queue))
-
+    keys = set(map(key, queue))
+    seen = dict(zip(keys, queue))
     while queue:
         current = queue.pop()
         yield current
         for node in get_neighbors(current):
-            if key(node) not in seen:
-                seen.add(key(node))
+            key_node = key(node)
```

Comment:
logic looks good, but we would not want this to run for all usages of this function.

 try to implement it without complicating the existing implementation (by giving functions as parameters where you want custom logic).